app/Misc/Router.php

```diff
@@ -12,12 +12,28 @@ class Router
      */
     public function __construct()
     {
-        $dispatcher = FastRoute\simpleDispatcher(function(FastRoute\RouteCollector $collector) {
-            $collector->addRoute(['GET'], '/logout', [new \App\Controllers\FrontController(), 'logout']);
-            $collector->addRoute(['GET', 'POST'], '/[{table}[/{action}[/{id}]]]', [new \App\Controllers\FrontController(), 'main']);
-        });
-
         $factory = new Abimo\Factory();
+        
+        $front_controller_class = $factory->config()->get('app','front_controller_class') ;
+        
+        $dispatcher = FastRoute\simpleDispatcher(
+          function(FastRoute\RouteCollector $collector) use ( $front_controller_class ) {
+            
+              $collector->addRoute(['GET'], '/logout', 
+                  function() use ($front_controller_class) { 
+                      return ( new $front_controller_class )->logout() ; 
```

Comment:
oh! right. 
the ::class keyword produces a string name of the class, it's the same result as enclosing it in single quotes, but this way I don't have to double the backslashes in the namespace path. 

It requires php 5.5+, but I picked it up from php-di:
http://php-di.org/doc/php-definitions.html#syntax